spec/api/v2/commercial/organizations_spec.rb

```diff
@@ -0,0 +1,244 @@
+      end
+    end
+  end
+
+  describe 'POST /api/v2/commercial/organization/update' do
+    context 'when params is missing' do
+      it 'renders an error' do
+        post '/api/v2/commercial/organization/update', headers: auth_header
+
+        expect_status_to_eq 422
+      end
+    end
+
+    context 'user is normal user' do
+      let(:do_request) do
+        post '/api/v2/commercial/organization/update',
+             headers: auth_header,
+             params: { oid: 'OID001' }
+      end
+      let(:test_user) { User.find(1) }
+
+      it 'deny request from normal user' do
+        do_request
+
+        expect(response.status).to eq(404)
+      end
+    end
+
+    context 'user is barong admin organization user' do
+      let(:params) { { oid: 'OID001' } }
+      let(:do_request) do
+        post '/api/v2/commercial/organization/update',
+             headers: auth_header,
+             params: params
+      end
+      let(:test_user) { User.find(1) }
+
+      it 'update organization field' do
+        # Assign user as barong organization admin
+        create(:membership, id: 1, user_id: 1, organization_id: 0)
+
+        params[:name] = 'Company Test'
+        do_request
+
+        expect(response.status).to eq(200)
+      end
+
+      it 'error when data not change' do
+        # Assign user as barong organization admin
+        create(:membership, id: 1, user_id: 1, organization_id: 0)
+
+        params[:name] = 'Company A'
+        do_request
+
+        expect(response.status).to eq(422)
+      end
+    end
+  end
+
+  describe 'POST /api/v2/commercial/organization/setting' do
+    context 'when params is missing' do
+      it 'renders an error' do
+        post '/api/v2/commercial/organization/setting', headers: auth_header
+
+        expect_status_to_eq 422
+      end
+    end
+
+    context 'user is normal user' do
+      let(:do_request) do
+        post '/api/v2/commercial/organization/setting',
+             headers: auth_header,
+             params: { oid: 'OID001' }
+      end
+      let(:test_user) { User.find(1) }
+
+      it 'deny request from normal user' do
+        do_request
+
+        expect(response.status).to eq(404)
+      end
+    end
+
+    context 'user is barong admin organization user' do
+      let(:params) { { oid: 'OID001' } }
+      let(:do_request) do
+        post '/api/v2/commercial/organization/setting',
+             headers: auth_header,
+             params: params
+      end
+      let(:test_user) { User.find(1) }
+
+      it 'update organization status' do
+        # Assign user as barong organization admin
+        create(:membership, id: 1, user_id: 1, organization_id: 0)
+
+        params[:status] = 'banned'
+        do_request
+
+        expect(response.status).to eq(200)
+      end
+
+      it 'update organization group' do
+        # Assign user as barong organization admin
+        create(:membership, id: 1, user_id: 1, organization_id: 0)
+
+        params[:group] = 'vip-1'
```

Comment:
what if unexisting fee group assigned

Reply:
This is no verify here for group. Do we need to get all group from `peatio` and validate at the backend?